### PR TITLE
armagetronad.*: 0.2.9.1.1 -> 0.2.9.2.3

### PR DESCRIPTION
**Pull request by numinit**
## Description of changes

Add `meta.mainProgram`, bump Git SHAs for unstable, and make sure that buildInputs is minimal for dedicated servers. Bump version to 0.2.9.2.3 due to upstream updates.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/armagetronad/default.nix [hunk 1]
```diff
@@ -154,20 +156,22 @@ let
...
+            lib.mapAttrs (name: value: mkArmagetron value dedicatedServer)
+            (lib.filterAttrs (name: value: (value dedicatedServer).version != (srcs.${latestVersionMajor} dedicatedServer).version) srcs)
+          ) //
+          {
+            # Allow both a "dedicated" passthru and a passthru for all the options other than the latest version, which this is.
+            dedicated = mkArmagetron fn true;
+          };
 
       meta = with lib; {
+        mainProgram = pname;
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:174]: The main point of mainProgram is to not rely on pname and be written out explicitly.

File: pkgs/games/armagetronad/default.nix [hunk 2]
```diff
@@ -154,20 +156,22 @@ let
...
         homepage = "http://armagetronad.org";
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:175]:
```suggestion
        homepage = "https://www.armagetronad.org/";
```

File: pkgs/games/armagetronad/default.nix [hunk 3]
```diff
@@ -28,64 +28,65 @@ let
...
       };
 
+    # https://gitlab.com/armagetronad/armagetronad/-/commits/hack-0.2.8-sty+ct+ap/?ref_type=heads
     "${latestVersionMajor}-sty+ct+ap" =
       let
-        rev = "fdfd5fb97083aed45467385b96d50d87669e4023";
-      in
-      {
+        rev = "a5bffe9dda2b43d330433f76f14eb374701f326a";
+        sha256 = "cNABxfg3MSmbxU/R78QyPOMwXGqJEamaFOPNw5yhDGE=";
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:70]:
```suggestion
        hash = "sha256-cNABxfg3MSmbxU/R78QyPOMwXGqJEamaFOPNw5yhDGE=";
```
please don't strip the prefixes of sri hashes

**Comment by numinit** [comment]
Should be ready.

**Comment by numinit** [comment]
@SuperSandro2000 I added a commit bumping version due to upstream updates and fixing the integration tests. A lot of them are failing to build with:

```
error:
       … from call site

         at «none»:0: (source not available)

       error: function 'anonymous lambda' called without required argument 'system'

       at /home/numinit/projects/nix/nixpkgs/nixos/tests/make-test-python.nix:1:4:

            1| f: {
             |    ^
            2|   system,
```

**Comment by SuperSandro2000** [comment]
Not sure what wrong there, how are you calling the tests. IIRC they should be run like ``nix-build -A nixosTests.vaultwarden``

**Comment by SuperSandro2000** [comment]
> Should be ready.

Does this still apply? Otherwise looks alright to me and we can merge it.

**Comment by numinit** [comment]
@SuperSandro2000 yeah, turns out I was doing `nix-build nixos/tests/armagetronad.nix`. Good from my end, all ready.

**Pull request merged**